### DISCUSSION TITLE
.toJSON() throws TypeError

### DISCUSSION THREAD

**Discussion by jaezudev** [discordjs/discord.js#6988] [Q&A] [answered]
### Issue description

When running below script, the line `commands.push(command.data.toJSON())` throws the following error: `TypeError: Cannot read properties of undefined (reading 'toJSON') at Object.<anonymous> (/storage/emulated/0/Projects/JS/pCounterBot/deploy-cmds-dev.js:13:29)`

How to reproduce:
1. Install packages `discord.js` and `@discordjs/rest`
1. Create discord bot profile, copy token and client id
2. Create test server, invite bot into there, copy guild id
3. Replace the variables with appropriate values
4. Save the script, navigate into folder containing the script
5. Run `node [your file name here].js`
6. Observe

Screenshot for visuals:
![Screenshot_20211115-140351_Termux](https://user-images.githubusercontent.com/66060223/141792181-0e8728b4-8496-4bc3-9b66-82f79ca15e4d.png)

### Code sample

```typescript
const fs = require('fs');
const { REST } = require('@discordjs/rest');
const { Routes } = require('discord-api-types/v9');
const clientId = 'client id here';
const guildId = 'guild id here';
const token = 'token here';

const commands = [];
const commandFiles = fs.readdirSync('./commands').filter(file => file.endsWith('.js'));

for (const file of commandFiles) {
	const command = require(`./commands/${file}`);
	commands.push(command.data.toJSON());
}

const rest = new REST({ version: '9' }).setToken(token);

rest.put(Routes.applicationGuildCommands(clientId, guildId), { body: commands })
	.then(() => console.log('Successfully registered application commands.'))
	.catch(console.error);
```

### discord.js version

13.3.1

### Node.js version

17.1.0

### Operating system

Linux, Windows

### Priority this issue should have

Medium (should be fixed soon)

### Which partials do you have configured?

No Partials

### Which gateway intents are you subscribing to?

GUILDS

**Comment by vladfrangu** [comment, marked as answer]
I don't see how this is related to discord.js. This is your code not exporting the data property of, what I assume is, an @discordjs/builder instance

**Reply by jaezudev** [reply]
Okay, I found out what the issue was. Thanks anyways for pointing out!

**Reply by TillVogelsang** [reply]
how can I fix it?

**Reply by GohstCamera** [reply]
> 

how to fond a issue ?
